### PR TITLE
[FIX] getAllReplies API Endpoint Test

### LINKED ISSUE #332: [TEST] getAllReplies API Endpoint
### Is your feature request related to a problem? Please describe.
Implement a test suite that will cover all required functionality expected from the getAllReplies endpoint `/v1/comments/{commentId}/replies`. For each test that doesn't pass, the controller method (along with its validation schema and swagger docs if the modifications require it) will be modified to include the required functionalities so that the test passes.

### Describe the solution you'd like
1. test that valid comment ID returns with the list of replies and status 200
2. test that valid comment ID filtered by ownerId returns with the filtered list response and status 200
3. test that valid comment ID filtered by isFlagged returns with the filtered list response and status 200
4. test that a valid comment ID with an unauthorized token returns with correct error response and status 401
5. test that invalid comment ID returns with correct error response and status 404

### Describe alternatives you've considered
None

### Additional context
None

### PR DESCRIPTION
**Pull request by simply-alliv**
**Before submitting your PR for review**

- Run `npm run lint` to find errors in code syntax/format
- Run `npm run lint:fix` to fix all auto-fixable errors in source code and auto-format with prettier
- Ensure you fix any linting errors displayed after running any of the above commands

**What does this PR do?**

Fixes #332 

This PR implements a test suite that will cover all required functionality expected from the getAllReplies endpoint /v1/comments/{commentId}/replies. For each test that doesn't pass, the controller method (along with its validation schema and swagger docs if the modifications require it) will be modified to include the required functionalities so that the test passes.

**description of Task to be completed?**

- [x] test that valid comment ID returns with the list of replies and status 200
- [x] test that valid comment ID filtered by ownerId returns with the filtered list response and status 200
- [x] test that valid comment ID filtered by isFlagged returns with the filtered list response and status 200
- [x] test that a valid comment ID with an unauthorized token returns with correct error response and status 401
- [x] test that invalid comment ID returns with correct error response and status 404

**How should this be manually tested?**

- In your terminal, run `npm test`

**Any background context you want to provide?**

None

**What is the link to the issue on Github?**

[Issue #332](https://github.com/microapidev/comment-microapi/issues/332)

**Questions:**

I am having trouble implementing the query for filtering by whether or not a reply has a flag or not for the isFlagged query parameter. I am hoping that someone with better knowledge on Mongo can help with making this query possible in an efficient way.

### CODE REVIEW

File: controllers/repliesController/getAllReplies.js [hunk 1]
```diff
@@ -31,9 +32,19 @@ const getAllReplies = async (req, res, next) => {
...
-    const replies = await Replies.find({ commentId: commentId }).populate(
-      "replyOwner"
-    );
+    // Create query for replies.
+    let query = {};
+
+    query.commentId = commentId;
+    if (ownerId) query.ownerId = ownerId;
+
+    if (!!isFlagged === true) {
```
Comment (piouson) [inline controllers/repliesController/getAllReplies.js:41]: you might be able to change this to:

```javascript
if (isFlagged) query["flags.0"] = {"$exists": true};
```

Comment (simply-alliv) [reply controllers/repliesController/getAllReplies.js:41]: Let me try that. Will get back to you with the outcome.

Comment (piouson) [reply controllers/repliesController/getAllReplies.js:41]: tested working on gitpod 
image

File: controllers/repliesController/getAllReplies.js [hunk 2]
```diff
@@ -38,10 +38,12 @@ const getAllReplies = async (req, res, next) => {
...
     if (ownerId) query.ownerId = ownerId;
 
-    if (!!isFlagged === true) {
-      query = { ...query, flags: { $size: { $gt: 1 } } };
-    } else if (!!isFlagged === false) {
-      query = { ...query, flags: { $size: { $eq: 0 } } };
+    if (typeof isFlagged === "string") {
+      if (isFlagged === "true") {
+        query["flags.0"] = { $exists: true };
+      } else if (isFlagged === "false") {
```
Comment (piouson) [inline controllers/repliesController/getAllReplies.js:44]: nice adaptation to falsy values!

**Pull request merged**